packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -0,0 +1,549 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/utils/EnumerableSet.sol";
+
+import "./UsingRegistry.sol";
+import "../common/Freezable.sol";
+import "../common/FixidityLib.sol";
+import "../common/Initializable.sol";
+
+import "../common/interfaces/IFeeHandler.sol";
+import "../common/interfaces/IFeeHandlerSeller.sol";
+
+// TODO move to IStableToken when it adds method getExchangeRegistryId
+import "../stability/StableToken.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
+import "../common/interfaces/ICeloToken.sol";
+import "../stability/interfaces/IExchange.sol";
+import "../stability/interfaces/ISortedOracles.sol";
+
+// Using the minimal required signatures in the interfaces so more contracts could be compatible
+import "../uniswap/interfaces/IUniswapV2RouterMin.sol";
+import "../uniswap/interfaces/IUniswapV2FactoryMin.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+
+contract FeeHandler is
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ICeloVersionedContract,
+  Freezable,
+  IFeeHandler,
+  ReentrancyGuard
+{
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  uint256 constant MAX_TIMESTAMP_BLOCK_EXCHANGE = 20;
+  uint256 public constant FIXED1_UINT = 1000000000000000000000000;
+
+  // Min units that can be burned
+  uint256 public constant MIN_BURN = 200;
+
+  // last day the daily limits were updated
+  uint256 public lastLimitDay;
+
+  FixidityLib.Fraction public burnFraction; // 80%
+
+  address public feeBeneficiary;
+
+  // This mapping can not be public because it contains  a FixidityLib.Fraction
+  // and that'd be only supported with experimental features in this
+  // compiler version
+  mapping(address => TokenState) private tokenStates;
+
+  struct TokenState {
+    address handler;
+    bool active;
+    FixidityLib.Fraction maxSlippage;
+    // Max amounts that can be burned in a day for a token
+    uint256 dailyBurnLimit;
```

Comment:
There is a discrepancy in naming of dailyBurnLimit and currentDaySellLimit. We should either use burn for both or sell for both